Sources/Plasma/NucleusLib/pnAsyncCoreExe/Private/Nt/pnAceNtSocket.cpp

```diff
@@ -255,46 +220,10 @@ static bool SocketDispatchRead (NtSock * sock) {
     ASSERT(sock->opRead.read.buffer == sock->buffer);
     ASSERT(sock->opRead.read.bytes);
 
-    // make sure there's an event procedure to handle this event
-    AsyncNotifySocketListen notify;
-    unsigned bytesProcessed;
-    sock->notifyProc = AsyncSocketFindNotifyProc(
-        sock->opRead.read.buffer,
-        sock->opRead.read.bytes,
-        &bytesProcessed,
-        &notify.connType, 
-        &notify.buildId,
-        &notify.buildType,
-        &notify.branchId,
-        &notify.productId
-    );
-    if (!sock->notifyProc)
-        return false;
-
-    // perform kNotifySocketListenSuccess
-    SocketGetAddresses(sock, &notify.localAddr, &notify.remoteAddr);
-    notify.param            = nullptr;
-    notify.asyncId          = nullptr;
-    notify.addr             = sock->addr;
-    sock->userState         = nullptr;
-    sock->connType          = notify.connType;
-    notify.buffer           = sock->opRead.read.buffer + bytesProcessed;
-    notify.bytes            = sock->opRead.read.bytes - bytesProcessed;
-    notify.bytesProcessed   = 0;
-    if (!sock->notifyProc((AsyncSocket) sock, kNotifySocketListenSuccess, &notify, &sock->userState))
-        return false;
-    bytesProcessed += notify.bytesProcessed;
-
-    // if we didn't use up all the bytes, dispatch a read operation
-    if (0 != (sock->opRead.read.bytes -= bytesProcessed)) {
-        sock->opRead.read.buffer += bytesProcessed;
-        if (!sock->notifyProc((AsyncSocket) sock, kNotifySocketRead, &sock->opRead.read, &sock->userState))
-            return false;
-    }
-
-    // add bytes used by IOsFindListenProc and kNotifySocketListenSuccess
-    sock->opRead.read.bytesProcessed += bytesProcessed;
-    return true;
+    // Probably unneeded since we just tested for nullptr above, but it
+    // maintains compatibility with what we had before...
+    sock->notifyProc = nullptr;
+    return false;
```

Comment:
I don't see where the deleted code nulls out the notifyProc? This looks a bit iffy IMO.

Reply:
Yeah, this one definitely needs some more clarification, since there were several steps involved in factoring it out...  Essentially:

* `s_notifyProcs` in pnAceIo.cpp was only ever read, not written to, meaning...
* `AsyncSocketFindNotifyProc` would never find an entry in `s_notifyProcs`, meaning...
* Whether `GetConnHash` fails or we look up an entry in `s_notifyProcs`, the return value of `AsyncSocketFindNotifyProc` is always `nullptr`, meaning...
* The call to `AsyncSocketFindNotifyProc` can be simplified to `nullptr`.

So you say, "But the call also had side effects!"...  Well, the side effects (setting values in `notify` and `bytesProcessed`) are only used if `sock->notifyProc` is non-null, which we've already proven is not the case.  Therefore, we always take the shortcut (`return false`) and the rest of the function is not used, meaning that even the call to `AsyncSocketFindNotifyProc` can be eliminated.

Reply:
OK, I think I understand. From my understanding, this was probably used on the server to register notify procedures for each connection type.